peps/pep-0767.rst

```diff
@@ -248,16 +251,16 @@ with ``ReadOnly`` is redundant, but it should not be seen as an error:
 Initialization
 --------------
 
-Assignment to a read-only attribute can only occur in the class declaring the attribute.
+Assignment to a read-only attribute can only occur in the class declaring the attribute,
+at sites described below.
 There is no restriction to how many times the attribute can be assigned to.
-Depending on the kind of the attribute, they can be assigned to at different sites:
 
 Instance Attributes
 '''''''''''''''''''
 
-Assignment to an instance attribute must be allowed in the following contexts:
+Assignment to a read-only instance attribute must be allowed in the following contexts:
```

Comment:
I think @oscarbenjamin raised an interesting point in the Discourse thread about this. The effects of `ReadOnly` as described in this PEP (structural typing and variance of generics) only require for soundness that we disallow _external_ mutation of the attribute. Disallowing _internal_ mutation (within methods of the class) is not required for soundness, and is arguably veering too far into `Final` territory.

It would simplify this section (and type checker implementations) quite a lot if we just specified that `ReadOnly` describes the external interface of the attribute, and all internal reassignments of a `ReadOnly` attribute (within methods of the class) are permitted.

Reply:
https://github.com/python/peps/pull/4559/commits/11f4522a750c3ef5bd7fe7c74f56e36963f93305